src/DetailsView/components/failure-instance-panel-control.tsx

```diff
@@ -133,39 +150,73 @@ export class FailureInstancePanelControl extends React.Component<FailureInstance
     private getFailureInstancePanelDetails = (): JSX.Element => {
         return (
             <FailureInstancePanelDetails
-                path={this.state.path}
-                snippet={this.state.snippet}
+                path={this.state.currentInstance.path}
+                snippet={this.state.currentInstance.snippet}
                 onSelectorChange={this.onSelectorChange}
                 onValidateSelector={this.onValidateSelector}
             />
         );
     };
 
+    private getCurrentInstance = (): FailureInstanceData => {
+        const updatedInstance = {
+            failureDescription: this.state.currentInstance.failureDescription,
+            path: this.state.currentInstance.path,
+            snippet: this.state.currentInstance.snippet,
+        };
+
+        return updatedInstance;
+    };
+
     protected onFailureDescriptionChange = (event, value: string): void => {
-        this.setState({ failureDescription: value });
+        const updatedInstance = this.getCurrentInstance();
```

Comment:
Why not just use the "this.state.currentInstance" here? Alternatively, if we can't use the state object directly, we can simply use a "clone" utility function here instead.